src/libsystemd-network/fuzz-dhcp-server.c

```diff
@@ -21,12 +21,15 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
         _cleanup_(sd_dhcp_server_unrefp) sd_dhcp_server *server = NULL;
         struct in_addr address = {.s_addr = htobe32(UINT32_C(10) << 24 | UINT32_C(1))};
         static const uint8_t chaddr[] = {3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3};
+        _cleanup_free_ uint8_t *duped = NULL;
         uint8_t *client_id;
         DHCPLease *lease;
 
         if (size < sizeof(DHCPMessage))
                 return 0;
 
+        assert_se(duped = memdup(data, size));
```

Comment:
What's weird is that the fuzz target has been running in its current form since 2018 and it seems to be the first time it has hit an issue like that. Looks like https://github.com/systemd/systemd/pull/22222 has changed something there. Even though the function signature doesn't say "const" anywhere I wonder if it's safe to assume that the dhcp server actually expects the data to be changed?

Reply:
> I wonder if it's safe to assume that the dhcp server actually expects the data to be changed?

Yes. The data is received by sd_dhcp_server, so it is ok to modify or update by itself. Also, the data is soon freed after dhcp_server_handle_message() is called.